app/src/main/java/com/google/android/sambadocumentsprovider/document/LoadDocumentTask.java

```diff
@@ -40,7 +40,11 @@ public LoadDocumentTask(Uri uri, SmbClient client, DocumentCache cache,
 
   @Override
   public DocumentMetadata run(Void... args) throws Exception {
-    return DocumentMetadata.fromUri(mUri, mClient);
+    if (mUri.getPathSegments().isEmpty()) {
```

Comment:
I don't think this piece of logic needs to be in a background thread, as it doesn't need any I/O.